src/core/ExpressionFormatter.ts

```diff
@@ -0,0 +1,383 @@
+import type { FormatOptions } from 'src/types';
+import { equalizeWhitespace } from 'src/utils';
+
+import InlineBlock from './InlineBlock';
+import Params from './Params';
+import { isReserved, type Token, TokenType } from './token';
+import {
+  AllColumnsAsterisk,
+  ArraySubscript,
+  AstNode,
+  BetweenPredicate,
+  BinaryClause,
+  Clause,
+  FunctionCall,
+  LimitClause,
+  Parenthesis,
+} from './ast';
+import { isTabularStyle } from './config';
+import Layout, { WS } from './Layout';
+import toTabularFormat, { isTabularToken } from './tabularStyle';
+
+interface ExpressionFormatterParams {
+  cfg: FormatOptions;
+  params: Params;
+  layout: Layout;
+  inline?: boolean;
+}
+
+/** Formats a generic SQL expression */
+export default class ExpressionFormatter {
+  private cfg: FormatOptions;
+  private inlineBlock: InlineBlock;
+  private params: Params;
+  private layout: Layout;
+
+  private inline = false;
+  private nodes: AstNode[] = [];
+  private index = -1;
+
+  constructor({ cfg, params, layout, inline = false }: ExpressionFormatterParams) {
+    this.cfg = cfg;
+    this.inline = inline;
+    this.inlineBlock = new InlineBlock(this.cfg.expressionWidth);
+    this.params = params;
+    this.layout = layout;
+  }
+
+  public format(nodes: AstNode[]): Layout {
+    this.nodes = nodes;
+
+    for (this.index = 0; this.index < this.nodes.length; this.index++) {
+      const node = this.nodes[this.index];
+      switch (node.type) {
```

Comment:
you can use the Enum in switch case with String()
ie. `switch (String(node.type)) { case: Enum.foo ... }`

Reply:
Not necessary, this works just fine:

```js
switch (node.type) {
    case: Enum.foo:
```